### PR TITLE
Open net cdf

### PR DESCRIPTION
**Pull request by lilyclements**
@africanmathsinitiative/developers this is ready to review

### CODE REVIEW

File: instat/dlgOpenNetCDF.vb [hunk 1]
```diff
@@ -16,15 +16,21 @@
 
 Imports instat.Translations
 Imports System.IO
+Imports RDotNet
 
 Public Class dlgOpenNetCDF
     Private bFirstLoad As Boolean = True
     Private bReset As Boolean = True
-    Private clsRDefaultFunction, clsRCDF, clsRClose As New RFunction
+    Private clsRDefaultFunction, clsRCDF, clsRClose, clsRFileDetails, clsRSubsetFunction, clsRLatFunction, clsRLongFunction, clsRZFunction, clsRTimeFunction As New RFunction
```
Comment (shadrackkibet) [inline instat/dlgOpenNetCDF.vb]: Naming convention?

**Comment by shadrackkibet** [comment]
@lilyclements would you mind explaining how this dialog works before i test it?

**Comment by lilyclements** [comment]
@shadrackkibet yes this is for testing netCDF files. I'll send you some via email

**Comment by shadrackkibet** [comment]
The dialog works fine but I am getting a lot of repeated code in my output window. Is this something to be addressed?`nc <- ncdf4::nc_open(filename="F:/AMI/Climatic data/ncdata.nc")
nc <- ncdf4::nc_open(filename="F:/AMI/Climatic data/ncdata.nc")
nc <- ncdf4::nc_open(filename="F:/AMI/Climatic data/ncdata.nc")
nc <- ncdf4::nc_open(filename="F:/AMI/Climatic data/ncdata.nc")
nc <- ncdf4::nc_open(filename="F:/AMI/Climatic data/ncdata.nc")
nc <- ncdf4::nc_open(filename="F:/AMI/Climatic data/ncdata.nc")
nc <- ncdf4::nc_open(filename="F:/AMI/Climatic data/ncdata.nc")
nc <- ncdf4::nc_open(filename="F:/AMI/Climatic data/ncdata.nc")
nc <- ncdf4::nc_open(filename="F:/AMI/Climatic data/ncdata.nc")
nc <- ncdf4::nc_open(filename="F:/AMI/Climatic data/ncdata.nc")
nc <- ncdf4::nc_open(filename="F:/AMI/Climatic data/ncdata.nc")
nc <- ncdf4::nc_open(filename="F:/AMI/Climatic data/ncdata.nc")
nc <- ncdf4::nc_open(filename="F:/AMI/Climatic data/ncdata.nc")
nc <- ncdf4::nc_open(filename="F:/AMI/Climatic data/ncdata.nc")
ncdf4::nc_close(nc=nc)
`

**Comment by lilyclements** [comment]
That's interesting and good to know. What have you been doing on the dialogs on R-Instat to get that?

**Comment by shadrackkibet** [comment]
I can't replicate exactly how i got this, will let you know In case i get the pattern.......

When does **short,medium,long** work? i am getting the message _Long/short/medium details not available_

**Comment by lilyclements** [comment]
So, that doesn't work for some of the data sets because it doesn't have that info., however, you can try it in the SIS Tamale dataset I sent you and that should work?
I also have another data set I can send you, however, this one is really big so I would not advise pressing OK on this one because it takes a very long time to run which is why I hadn't sent it before. A "subset" feature is being worked on so that we can use these larger data sets.

**Comment by dannyparsons** [comment]
I'm going to merge this as I have edited the method for this on another branch. But I will send it back up soon for continued testing. We are using this dialog next week at a workshop so it will needs lots of testing before then.

**Pull request merged**